src/code-considerations/criteria-for-inclusion.md

```diff
@@ -0,0 +1,70 @@
+# Criteria for inclusion in `std`
+
+**Note**: *The process for deciding what does and does not get added to the standard
+library is too complex to reduce to a set of concrete rules. Ultimately we rely
+upon the expertise of members of the library API team, but here we attempt to
+document the reasoning that has been used historically to guide contributors
+and team members and to ensure long term consistency in how these decisions are
+made even as the team membership naturally changes over time.*
+
+## Reasons For Inclusion
+
+* APIs where there is a clear best implementation
```

Comment:
I would add "General purpose APIs likely to be useful to many users" or similar text. E.g., one could imagine an API which ticks all these boxes, but if it is only useful for games programmers then we probably don't want it.

Reply:
I think that would depend on circumstances; we may be willing to add APIs exclusively useful to a subset of users. For instance, portable SIMD, or miscellaneous OS-specific APIs. And there are probably *fewer* people using those than there are game programmers.

Reply:
Maybe "many kinds of users" rather than "many users"? Like SIMD feels more generally useful than say OpenGL support, even if the latter might have more users.

Reply:
Maybe OpenGL is a bad example here. Maybe something specialised like text layout or something? I'm sure there are crates which would be better be kept as crates simply because they're unlikely to be used by a broad slice of users?

Put it another way, I agree with you that it "would depend on circumstances" but it seems we want some guideline about APIs being general purpose as a starting point.